python/cudf/cudf/io/orc.py

```diff
@@ -230,35 +247,58 @@ def read_orc(
 
     from cudf import DataFrame
 
-    is_single_filepath_or_buffer = ioutils.ensure_single_filepath_or_buffer(
-        path_or_data=filepath_or_buffer, **kwargs,
-    )
-    if not is_single_filepath_or_buffer:
-        raise NotImplementedError(
-            "`read_orc` does not yet support reading multiple files"
-        )
+    # Multiple sources are passed as a list. If a single source is passed,
+    # wrap it in a list for unified processing downstream.
+    if not is_list_like(filepath_or_buffer):
+        filepath_or_buffer = [filepath_or_buffer]
+
+    # Each source must have a correlating stripe list. If a single stripe list
+    # is provided rather than a list of list of stripes then extrapolate that
+    # stripe list across all input sources
+    if stripes is not None:
+        if any(not isinstance(stripe, list) for stripe in stripes):
+            stripes = [stripes]
+
+        # Must ensure a stripe for each source is specified, unless None
+        if len(stripes) is not len(filepath_or_buffer):
```

Comment:
I think `is not` is too strong a qualification as opposed to `if not len(stripes) == len(filepath_or_buffer):`